lightning/src/ln/channelmanager.rs

```diff
@@ -3547,12 +3547,15 @@ where
 				channel_funding_txo: shutdown_res.channel_funding_txo,
 			}, None));
 
-			let funding_info = if is_manual_broadcast {
-				shutdown_res.channel_funding_txo.map(|outpoint| FundingInfo::OutPoint{ outpoint })
-			} else {
-				shutdown_res.unbroadcasted_funding_tx.map(|transaction| FundingInfo::Tx{ transaction })
-			};
-			if let Some(funding_info) = funding_info {
+			if let Some(transaction) = shutdown_res.unbroadcasted_funding_tx {
```

Comment:
Would it be easy to add test coverage for the bug described in the commit message (maybe in follow-up)? Seems obvious in retrospect, ugh.